packages/nextjs/app/page.tsx

```diff
@@ -1,16 +1,36 @@
 "use client";
 
+import { useEffect, useState } from "react";
 import Link from "next/link";
 import type { NextPage } from "next";
+import { usePublicClient } from "wagmi";
 import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
-import { useScaffoldContractRead } from "~~/hooks/scaffold-eth";
+import { Address } from "~~/components/scaffold-eth";
+import { useScaffoldContractRead, useScaffoldEventHistory } from "~~/hooks/scaffold-eth";
 
 const Home: NextPage = () => {
+  const [checkedInBuilders, setCheckedInBuilders] = useState<string[]>([]);
+  const [isConnectedOptimism, setIsConnectedOptimism] = useState<boolean>(false);
+
+  //HOOKS
+
+  const { chain } = usePublicClient();
+
+  //Get tally of Checkedin Builders
   const { isLoading, data: checkedInCounter } = useScaffoldContractRead({
     contractName: "BatchRegistry",
     functionName: "checkedInCounter",
   });
 
+  //event history
+  const { isLoading: isReadingEventLoading, data: eventHistory } = useScaffoldEventHistory({
+    contractName: "BatchRegistry",
+    eventName: "CheckedIn",
+    fromBlock: 116978463n,
+    watch: true,
+    enabled: isConnectedOptimism,
```

Comment:
We could remove both this, we don't need to `watch` since `CheckedIn` event will change occasionally, adding `watch` will fetch every 30 sec checkout [this](https://docs.scaffoldeth.io/hooks/useScaffoldEventHistory#configuration)  and puts loads on RPC 😅